spaceship/lib/spaceship/connect_api/models/app_screenshot.rb

```diff
@@ -111,7 +111,9 @@ def self.create(app_screenshot_set_id: nil, path: nil, wait_for_processing: true
           timeout_minutes = (ENV["SPACESHIP_SCREENSHOT_UPLOAD_TIMEOUT"] || 20).to_i
 
           loop do
-            puts("Waiting for screenshot to appear before uploading...")
+            # This error handling needs to be resived since any error occured can reach here.
+            # It should handle errors based on what status code is.
+            puts("Waiting for screenshots to appear before uploading. This is unlikely to be recovered unless it's 503 error. error=\"#{error}\"")
```

Comment:
This is to give us more details when fastlane-users report issues. I feel like we should revise this error handling to check error response's status code to be accurate (or maybe we should get rid of this workaround somehow...?) 

See https://github.com/fastlane/fastlane/issues/17243#issuecomment-700262947